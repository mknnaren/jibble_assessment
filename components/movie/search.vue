<template>
  <div>
    <div class="text-center pt-2">
      <v-text-field
        outlined
        label="Search"
        v-model="searchText"
        @change="showPageResults"
        prepend-inner-icon="mdi-magnify"
        :append-icon="searchText.length > 0 ? 'mdi-close-circle' : ''"
        @click:append="searchText=''"
      ></v-text-field>
    </div>
  </div>
</template>
<script lang="ts">
import { defineComponent, watch, ref } from "@nuxtjs/composition-api";
export default defineComponent({
  emits: ["show-search-list"],
  setup(props, context) {
    const searchText = ref("");
    function showPageResults() {
      context.emit("show-search-list", searchText.value);
    }
    
    watch(searchText, showPageResults);
    return {
      searchText,
      showPageResults
    };
  },
});
</script>
