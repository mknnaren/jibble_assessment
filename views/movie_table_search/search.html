<div>
    <div class="text-center pt-2">
        <v-text-field outlined label="Search" 
        v-model="searchText" 
        @change="showPageResults" 
        prepend-inner-icon="mdi-magnify" 
        :append-icon="searchText.length>0 ? 'mdi-close-circle' : ''" 
        @click:append="clearMessage"
        ></v-text-field>
    </div>
</div>

