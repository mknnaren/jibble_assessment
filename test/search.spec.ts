import { mount, createLocalVue  } from '@vue/test-utils'
import search from '@/components/movie/search.vue';
import Vuetify from 'vuetify';
import Vue from 'vue'
Vue.use(Vuetify)

describe('search', () => {
  let localVue = createLocalVue()
  let vuetify = new Vuetify()
  test('mounts properly', () => {
    const wrapper = mount(search, {localVue, vuetify})
    expect(wrapper.vm).toBeTruthy()
  })

  test('search result event emited properly', () => {
    const wrapper = mount(search, {localVue, vuetify})
    const input = wrapper.find('input')
    input.trigger('focus')
    // input.element.value = 'h'
    input.trigger('input')
    input.trigger('keydown.enter')
    input.trigger('keydown.enter')
    expect(wrapper.emitted()["show-search-list"]).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = mount(search,  {localVue, vuetify})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
