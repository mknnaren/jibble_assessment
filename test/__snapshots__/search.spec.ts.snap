// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search renders properly 1`] = `
"<div>
  <div class=\\"text-center pt-2\\">
    <div class=\\"v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-input__prepend-inner\\">
            <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-magnify theme--light\\"></i></div>
          </div>
          <fieldset aria-hidden=\\"true\\">
            <legend style=\\"width: 0px;\\"><span>â€‹</span></legend>
          </fieldset>
          <div class=\\"v-text-field__slot\\"><label for=\\"input-14\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Search</label><input id=\\"input-14\\" type=\\"text\\"></div>
        </div>
        <div class=\\"v-text-field__details\\">
          <div class=\\"v-messages theme--light\\">
            <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
